_rules/role-required-states-and-properties-4e8ab6.md

```diff
@@ -50,9 +48,9 @@ This rule relies on browsers and assistive technologies to support leaving out [
 
 ## Background
 
-Omitting required ARIA properties is often the result of a developer error. When required properties are missing some browsers and assistive technologies will guess the property, or leave the element inaccessible.
+Omitting [WAI-ARIA required states and properties][] is often the result of a developer error. When required properties are missing and a default value is not specified by [WAI-ARIA Specifications](#wai-aria-specifications), the behavior is not defined. For [WAI-ARIA 1.2][], the only [explicit semantic roles][explicit semantic role] with a required property with a default value are the `option` and `tabs roles` for the `aria-selected` property.
 
-This rule is testing author built components, not user-agent built ones. Elements that keep their [implicit semantic role][] are mapped into conforming accessible objects, with all required properties, by user agents and are therefore not tested by this rule. Most of these mappings are defined in the [HTML Accessibility API Mappings, Attribute State and Property Mappings](https://www.w3.org/TR/html-aam-1.0/#html-attribute-state-and-property-mappings).
+This rule is testing author built components that specify [explicit semantic roles][explicit semantic role] and not components that keep their [implicit semantic role][]. For components that keep their [implicit semantic role][], all native HTML and SVG elements have native attributes that are mapped to all of the [WAI-ARIA required states and properties](https://www.w3.org/TR/wai-aria/#requiredState). Most of these mappings are defined in the [HTML Accessibility API Mappings, Attribute State and Property Mappings](https://www.w3.org/TR/html-aam-1.0/#html-attribute-state-and-property-mappings).
```

Comment:
**Nitpick**
```suggestion
This rule is testing author built components that specify [explicit semantic roles][] and not components that keep their [implicit semantic role][]. For components that keep their [implicit semantic role][], all native HTML and SVG elements have native attributes that are mapped to all of the [WAI-ARIA required states and properties][]. Most of these mappings are defined in the [HTML Accessibility API Mappings, Attribute State and Property Mappings](https://www.w3.org/TR/html-aam-1.0/#html-attribute-state-and-property-mappings).
```
I also tend to prefer using references list rather than inline links, i.e., also replace
```
[HTML Accessibility API Mappings, Attribute State and Property Mappings](https://www.w3.org/TR/html-aam-1.0/#html-attribute-state-and-property-mappings)
```
by
```
[HTML Accessibility API Mappings, Attribute State and Property Mappings][aria attribute mappings]
```
and add, at the end of the file, an entry:
```
[aria attribute mappings]: https://www.w3.org/TR/html-aam-1.0/#html-attribute-state-and-property-mappings 'HTML AAM, definition of ARIA attributes mappings'
```

Reply:
Perhaps I misunderstand how the glossary works. In your suggested change of `[explicit semantic roles][explicit semantic role]` to `[explicit semantic roles][]`, the key in the glossary is `explicit semantic role` (without the trailing s). I didn't originally that bit of content, but I assumed that since the first part didn't match the glossary key that the second part was required?